// react-skycons/lib/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

/*! *****************************************************************************
Copyright (c) Microsoft Corporation.

Permission to use, copy, modify, and/or distribute this software for any
purpose with or without fee is hereby granted.

THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
PERFORMANCE OF THIS SOFTWARE.
***************************************************************************** */

var __assign = function() {
    __assign = Object.assign || function __assign(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};

function __rest(s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
}

var globalThis = require('globalthis')();
var Skycons = require('skycons')(globalThis);
(function (SkyconsType) {
    SkyconsType["CLEAR_DAY"] = "CLEAR_DAY";
    SkyconsType["CLEAR_NIGHT"] = "CLEAR_NIGHT";
    SkyconsType["PARTLY_CLOUDY_DAY"] = "PARTLY_CLOUDY_DAY";
    SkyconsType["PARTLY_CLOUDY_NIGHT"] = "PARTLY_CLOUDY_NIGHT";
    SkyconsType["CLOUDY"] = "CLOUDY";
    SkyconsType["RAIN"] = "RAIN";
    SkyconsType["SLEET"] = "SLEET";
    SkyconsType["SNOW"] = "SNOW";
    SkyconsType["WIND"] = "WIND";
    SkyconsType["FOG"] = "FOG";
})(exports.SkyconsType || (exports.SkyconsType = {}));
function SkyconsComponent(props) {
    var color = props.color, _a = props.animate, animate = _a === void 0 ? true : _a, resizeClear = props.resizeClear, type = props.type, style = props.style, _b = props.size, size = _b === void 0 ? 24 : _b, restPops = __rest(props, ["color", "animate", "resizeClear", "type", "style", "size"]);
    var canvasRef = React.useRef(null);
    React.useEffect(function () {
        var skycons = new Skycons({
            color: color,
            resizeClear: resizeClear,
        });
        skycons.add(canvasRef.current, Skycons[type]);
        if (animate) {
            skycons.play();
        }
        return function () {
            skycons.pause();
            skycons.remove();
        };
    }, [animate, color, type, resizeClear]);
    return (React__default['default'].createElement("canvas", __assign({ ref: canvasRef, width: size, height: size, style: __assign({ width: size, height: size }, style), "aria-label": type, role: "img" }, restPops)));
}

exports.Skycons = SkyconsComponent;
exports.default = SkyconsComponent;
